src/script/lua_api/l_metadata.cpp

```diff
@@ -229,6 +229,56 @@ int MetaDataRef::l_get_keys(lua_State *L)
 	return 1;
 }
 
+// get_bool(self, name)
+int MetaDataRef::l_get_bool(lua_State *L)
+{
+	MAP_LOCK_REQUIRED;
+
+	MetaDataRef *ref = checkAnyMetadata(L, 1);
+	std::string name = luaL_checkstring(L, 2);
+
+	IMetadata *meta = ref->getmeta(false);
+	if (meta == NULL) {
+    	return luaL_error(L, "The boolean meta value can't be nil");
+	}
+
+	std::string str_;
+	const std::string &str = meta->getString(name, &str_);
+
+	// Explicit check for "true" or "false"
+	if (str == "true") {
+		lua_pushboolean(L, true);
+	} else if (str == "false") {
+		lua_pushboolean(L, false);
+	} else {
+		// Throw error if the value is neither "true" nor "false"
+		return luaL_error(L, "Invalid metadata value. Expected 'true' or 'false'.");
+	}
+	return 1;
+}
+
+// set_bool(self, name, var)
+int MetaDataRef::l_set_bool(lua_State *L)
+{
+	MAP_LOCK_REQUIRED;
+
+	MetaDataRef *ref = checkAnyMetadata(L, 1);
+	std::string name = luaL_checkstring(L, 2);
+
+	// Ensure only boolean values are accepted
+	if (!lua_isboolean(L, 3)) {
+		return luaL_error(L, "Invalid argument type. Expected boolean.");
+	}
+
+	int a = lua_toboolean(L, 3);
+	std::string str = a ? "true" : "false";
+
+	IMetadata *meta = ref->getmeta(true);
```

Comment:
when might `meta` be null? Feels like there should be a `SANITY_CHECK` or at least an `assert` here.

Reply:
You’d hope never, however there is always a chance that someone makes a hack that can make it null.